Release notes for SDK v0.28.0 (#2798)
wandb/docs: release-notes/sdk-releases.mdx

This page includes release notes for the W&B Python SDK (`wandb` package) used with W&B Models. For the Weave Python SDK (`weave` package), see [Weave SDK releases](/release-notes/weave-sdk-releases). For W&B Server, see [W&B Server release notes](/release-notes/server-releases).

<Update label="0.28.0" description="June 22, 2026">

> ## Notable Changes

Author: ```suggestion
## Notable changes
```

- This version drops compatibility with server versions older than 0.65.0.

## Added
- W&B LEET TUI features:
  - Media panels support high-resolution image rendering in terminals that support the Kitty protocol, with ANSI fallback.
  - Media panels support synced scrubbing: press `l` to link scrubbing, then navigate with arrow keys, Home, and End, which move a shared cursor over the union step timeline and every image tile follows it.
  - Basic remote-run support with `wandb leet [run-url]`.
- The following paginated artifacts and registry API methods now accept an optional order string as a keyword argument: `Api.artifacts()`, `Api.artifact_collections()`, `Api.registries()`, `Api.registries().collections()`, `Registry.collections()`.

## Changed
- W&B LEET TUI changes:
  - The W&B LEET TUI is now generally available, and the command is updated from `wandb beta leet` to `wandb leet`. The prior command is still available as an alias.

>   - Lists of images logged under a single key are now displayed in the W&B LEET media pane, one tile per image (@dmitryduev in #12033)

Reviewer: ```suggestion
  - Lists of images logged under a single key are now displayed in the W&B LEET media pane, one tile per image.
```

- To improve query performance, `wandb.Api().runs()` no longer loads Sweeps for each run by default. Sweep data is loaded on first access of the sweep property.

## Fixed

> - `File.download()` no longer fails after a hard-coded 5-second timeout; downloads go through `wandb-core` and respect the file transfer setting.

Reviewer: ```suggestion
- `File.download()` no longer fails after a hard-coded 5-second timeout; downloads respect the file transfer settings.
```

- Fixed a regression introduced in 0.27.1 where `wandb.Api().viewer`, `Api().user()`, and `Api().users()` could fail with the error `WandbApiFailedError: relogin required for some API keys`.

> - When a `wandb.Image` is logged inside a `wandb.Table` with multiple box or mask layers with distinct `class_labels`, each layer's labels are now preserved in new `box_class_maps` or `mask_class_maps` fields in the `table.json`. Previously, a single `class_map` would be incorrectly overwritten by each set of class labels. An upcoming Server release will contain a corresponding UI fix to read these per-layer fields. Legacy servers will retain the current behavior.

Reviewer: > An upcoming Server release will contain a corresponding UI fix to read these per-layer fields. Legacy servers will retain the current behavior.

Shall we talk about the future like that??

Author: Well probably not, but y'all did add it to your changelog.

> - When a `wandb.Image` is logged inside a `wandb.Table` with multiple box or mask layers with distinct `class_labels`, each layer's labels are now preserved in new `box_class_maps` or `mask_class_maps` fields in the `table.json`. Previously, a single `class_map` would be incorrectly overwritten by each set of class labels. An upcoming Server release will contain a corresponding UI fix to read these per-layer fields. Legacy servers will retain the current behavior.

Author: ```suggestion
- When a `wandb.Image` is logged inside a `wandb.Table` with multiple box or mask layers with distinct `class_labels`, each layer's labels are now preserved in new `box_class_maps` or `mask_class_maps` fields in the `table.json`. Previously, a single `class_map` would be incorrectly overwritten by each set of class labels.
```
